samples/cpp/common/utils/src/latency_metrics.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// clang-format off
+#include <algorithm>
+#include <map>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "samples/latency_metrics.hpp"
+// clang-format on
+
+void LatencyMetrics::write_to_stream(std::ostream& stream) const {
+    std::ios::fmtflags fmt(std::cout.flags());
+    stream << data_shape << ";" << std::fixed << std::setprecision(2) << median_or_percentile << ";" << avg << ";"
+           << min << ";" << max;
+    std::cout.flags(fmt);
+}
+
+void LatencyMetrics::write_to_slog() const {
+    std::string percentileStr = (percentile_boundary == 50)
+                                    ? "\tMedian:     "
+                                    : "\t" + std::to_string(percentile_boundary) + " percentile:    ";
+    if (!data_shape.empty()) {
+        slog::info << "\tData shape: " << data_shape << slog::endl;
+    }
+    slog::info << percentileStr << double_to_string(median_or_percentile) << " ms" << slog::endl;
+    slog::info << "\tAverage:    " << double_to_string(avg) << " ms" << slog::endl;
+    slog::info << "\tMin:        " << double_to_string(min) << " ms" << slog::endl;
+    slog::info << "\tMax:        " << double_to_string(max) << " ms" << slog::endl;
+}
+
+const nlohmann::json LatencyMetrics::to_json() const {
+    nlohmann::json stat;
+    stat["data_shape"] = data_shape;
+    stat["latency_median"] = median_or_percentile;
+    stat["latency_average"] = avg;
+    stat["latency_min"] = min;
+    stat["latency_max"] = max;
+    return stat;
```

Comment:
If we will hide BA in future, I don't want to have json library in samples. 
BA needs json to read config, in order to write json - we can write our simple code, is not it?

Reply:
samples don't need json at all. I refactored it